Content.Shared/Damage/Components/DamageableComponent.cs

```diff
@@ -18,73 +18,92 @@ namespace Content.Shared.Damage.Components
+        /// <summary>
+        ///     The main damage dictionary. All the damage information is stored in this dictionary with <see cref="DamageTypePrototype"/>  keys.
+        /// </summary>
+        private Dictionary<DamageTypePrototype, int> _damageDict = new();
 
         [DataField("resistances")]
-        public string ResistanceSetId { get; set; } = DefaultResistanceSet;
+        public string ResistanceSetId { get; set; } = "defaultResistances";
 
         [ViewVariables] public ResistanceSet Resistances { get; set; } = new();
 
         // TODO DAMAGE Use as default values, specify overrides in a separate property through yaml for better (de)serialization
         [ViewVariables]
         [DataField("damageContainer")]
-        public string DamageContainerId { get; set; } = DefaultDamageContainer;
+        public string DamageContainerId { get; set; } = "metallicDamageContainer";
 
         // TODO DAMAGE Cache this
-        [ViewVariables] public int TotalDamage => _damageList.Values.Sum();
-        [ViewVariables] public IReadOnlyDictionary<DamageGroupPrototype, int> DamageClasses => damageListToDamageGroup(_damageList);
-        [ViewVariables] public IReadOnlyDictionary<DamageTypePrototype, int> DamageTypes => _damageList;
+        // When moving logic from damageableComponent --> Damage System, make damageSystem update these on damage change.
+        [ViewVariables] public int TotalDamage => _damageDict.Values.Sum();
+        [ViewVariables] public IReadOnlyDictionary<DamageTypePrototype, int> GetDamagePerType => _damageDict;
+        [ViewVariables] public IReadOnlyDictionary<DamageGroupPrototype, int> GetDamagePerApplicableGroup => DamageTypeDictToDamageGroupDict(_damageDict, ApplicableDamageGroups);
+        [ViewVariables] public IReadOnlyDictionary<DamageGroupPrototype, int> GetDamagePerFullySupportedGroup => DamageTypeDictToDamageGroupDict(_damageDict, FullySupportedDamageGroups);
+
+        // Whenever sending over network, also need a <string, int> dictionary
+        // TODO DAMAGE MAYBE Cache this?
+        public IReadOnlyDictionary<string, int> GetDamagePerApplicableGroupIDs => ConvertDictKeysToIDs(GetDamagePerApplicableGroup);
+        public IReadOnlyDictionary<string, int> GetDamagePerFullySupportedGroupIDs => ConvertDictKeysToIDs(GetDamagePerFullySupportedGroup);
+        public IReadOnlyDictionary<string, int> GetDamagePerTypeIDs => ConvertDictKeysToIDs(_damageDict);
+
+        // TODO PROTOTYPE Replace these datafield variables with prototype references, once they are supported.
+        // Also requires appropriate changes in OnExplosion() and RadiationAct()
+        [ViewVariables]
+        [DataField("radiationDamageTypes")]
+        public List<string> RadiationDamageTypeIDs { get; set; } = new() {"Radiation"};
+        [ViewVariables]
+        [DataField("explosionDamageTypes")]
+        public List<string> ExplosionDamageTypeIDs { get; set; } = new() { "Piercing", "Heat" };
```

Comment:
blech but i wont have you change this now

Reply:
Yeah....

I think clyf on discord was going to rework radiation, and explosion eventually has to be redone as well.
I still want to make these damage groups, but would do that in a separate PR.